File: examples/javascript/client/proto/ReqSceneEnterFly.js
module.exports = ReqSceneEnterFly;

var Packet = require('../net/Packet');


function ReqSceneEnterFly() {
	this._map_id = undefined;


	this.Encode = function() {
		var packet = new Packet();
		packet.WriteUint(this._map_id);
		packet.Encode(Msg.REQ_SCENE_ENTER_FLY);
		return packet;
	}


	this.SetMapId = function(map_id) {
		this._map_id = map_id;
	}
	this.GetMapId= function() {
		return this._map_id;
	}
}

File: examples/golang/client/proto/req_scene_enter_fly.go
package proto

import (
	"packet"
)

type ReqSceneEnterFly struct {
	MapId                    uint32
}

func (this *ReqSceneEnterFly) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint32(this.MapId)

	return pack.Encode(P_REQ_SCENE_ENTER_FLY)
}
